### PR TITLE
Add enhanced lint security checks. Bump CI toolkit to 3.7.1. 

### PR DESCRIPTION
**Pull request by wzieba**
### Description

This PR integrates enhanced security Android lint checks: https://github.com/google/android-security-lints. 

Additionally, it fixes not working Android SARIF file upload on main branch.

### Testing

One can compare lint report before and after this PR. The report from this PR reports more lint issues in `Security` category.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: config/lint/lint.xml
Comment (wzieba) [inline config/lint/lint.xml:47]: These issues are warnings by default. As none of them is considered an error (meaning: we should fix this ASAP), I decided to simply mark them as such and delegate addressing them to the product team.

Comment (ParaskP7) [reply config/lint/lint.xml:47]: @wzieba , do you mean you did the same trick as done with [UsingMaterialAndMaterial3Libraries](https://github.com/wordpress-mobile/WordPress-Android/blob/da3176184abdce2e40838e5a92e4c5c5b0e9e67b/config/lint/lint.xml#L74-L75) so as to not fail the build, but also to not ignore them, have them available on the report, right? 🤔 

If that's the case I think we better create an issue per these new Lint issues, just like it was done [here](https://github.com/wordpress-mobile/WordPress-Android/issues/21065) and inform @nbradbury about it too, just so that he's aware of those and prioritizes those security issues accordingly.

Comment (ParaskP7) [reply config/lint/lint.xml:47]: And just as an extra FYI here, we really shouldn't forget to remove these issues from `lint.xml` when we fix each of them. Otherwise, any new such issues will be ignored and just added on the report, that is, instead of having Lint failing on the spot and forcing the developer to correct the issue right there, before merging that new "vulnerable" PR.

**Review by ParaskP7 — APPROVED** [review]
👋 @wzieba !

I have reviewed this PR as per the instructions, great checks addition, Lint security for the win, for (almost) free! 🎉

Suggestion (💡): As per my comment [here](https://github.com/wordpress-mobile/WordPress-Android/pull/21251/#discussion_r1772957607) I suggest creating new issues per Lint issue and adding a `TODO` comment on top of them, just so that we don't forget to remove this configuration later. Having said that, I am still approving this PR so as not to block it in any case.

File: config/lint/lint.xml
Comment (wzieba) [reply config/lint/lint.xml:47]: I see your point and added a comment (f8ea724da24d36ece9c65b517f7a178cf6ec5044). 

Generally, I think we should revisit our strategy on Android lint, as other projects doesn't have `warningsAsErrors` configuration, and it works for them. I'm not sure if we should treat WordPress-Android differently 🤔

**Comment by wzieba** [comment]
> As per my comment https://github.com/wordpress-mobile/WordPress-Android/pull/21251/#discussion_r1772957607 I suggest creating new issues per Lint issue and adding a TODO comment on top of them, just so that we don't forget to remove this configuration later.

I've added the TODO comment, but I'm hesitent about adding a new GitHub issue. As we use GitHub Code Scanning feature, I think it'll be just duplication of work, making it all more complex than it has to be. 

For the list of lint issues, one can visit https://github.com/wordpress-mobile/WordPress-Android/security/code-scanning and, even better, sort it by category and severity, which works better than GitHub issues.

File: config/lint/lint.xml
Comment (ParaskP7) [reply config/lint/lint.xml:47]: > I see your point and added a comment (https://github.com/wordpress-mobile/WordPress-Android/commit/f8ea724da24d36ece9c65b517f7a178cf6ec5044).

I am not sure if this comment is enough @wzieba , that is, without a TODO and an issue linked, but I am leaving this up to @nbradbury to decide. 🤷 

> Generally, I think we should revisit our strategy on Android lint, as other projects doesn't have warningsAsErrors configuration, and it works for them. I'm not sure if we should treat WordPress-Android differently 🤔

I started with `warningsAsErrors` with JP/WPAndroid and it seems to have been working just fine for more than a year now, no new warnings were added and the baseline issues kept dropping to just about 300 atm (we started with thousands of them). I just never had time to do the same on other projects as well. I am not sure why you think it is not a step forward for other projects to do so, to me it is pretty clear that it is. 🤔

Granted, we could now have the `sarif` process that will add a comment on a PR for every new warning/error, but this doesn't enforce developers to fix the warning, they can simply ignore it, just like they do now, and I would argue that in make cases this is exactly what will happen. On the other hand, with `warningsAsErrors` you can't just ignore it, you either fix it or suppress it (you should provide a reason too). A reviewer should then check the fix or suppression and (dis)agree on it. Also, with `warningsAsErrors` you get a quick feedback on it locally too, you don't need to depend on CI running Lint and create that comment for you, so you can fix the warning even before pushing/creating a PR. Of course, you need to run the `lint` task to get that local feedback, and granted not many developers do that, but I (usually) do, especially when I was developing new features working on a product.

Comment (ParaskP7) [reply config/lint/lint.xml:47]: Responding to [this](https://github.com/wordpress-mobile/WordPress-Android/pull/21251#issuecomment-2370775753) here:

> I've added the TODO comment, but I'm hesitent about adding a new GitHub issue. As we use GitHub Code Scanning feature, I think it'll be just duplication of work, making it all more complex than it has to be.

For me this is about making this visible to product engineers so that they handle it at some point, not about duplication. Making this also visible from the IDE itself, having the `TODO` there with a link is way easier than to have to know what code scanning is and navigate to it to figure out the rest. 🤔 

> For the list of lint issues, one can visit https://github.com/wordpress-mobile/WordPress-Android/security/code-scanning and, even better, sort it by category and severity, which works better than GitHub issues.

Yeap, true, but I am not sure how easy is to prioritize tech debt with code scanning, versus handling a self-created issue, which could hold even more details and maybe a link to code scanning as well. 🤷

Comment (wzieba) [reply config/lint/lint.xml:47]: I see your points 👍 But I also think, there's a reason that some issues are considered `warnings` and some of them are `errors` by their authors, so I'm not sure if `warningsAsErrors` is _the_ best strategy to follow.

Comment (ParaskP7) [reply config/lint/lint.xml:47]: Yeap, true, I do see your point too, and if I were to speak with another version of myself, I would agree, but if I did agree with that, then I would argue that `Detekt` and other such code analysis tools, too, should be considered optional. Yet, we don't consider them optional and instead we fail the build, even on unused imports, which can be considered `informational`, not even a `warning`. 🤷 

Thus, in my mind, this is not a discussion about `warnings` versus `errors` but more so about code quality overall, and what we can do to effortlessly enforce that. When I say effortlessly, I mean why not block developer from using merging code that warns then on `UseAppTint`, which is a simple use `app:tint` instead of `android:tint` fix. Why ignore that warning if it can be quickly solved. That's all I am saying here... 😊

**Pull request merged**
